### PR TITLE
More changes: Web examples

### PR DESCRIPTION
**Pull request by kassane**
**changes:**
* http server added
* update http/tls client (kyber)
* refactoring build-system
* CI - Windows added

cc: @Panquesito7 

**Note:** zig-tests don't stop to print `stderr` (aka `std.debug.print`).

### CODE REVIEW

**Comment by kassane** [comment]
`Http.Server` tests on Windows and MacOS has issue to get request, linux works. Please see the last action!
@Panquesito7

---
edit

@truemedian , any suggestions about?
Trying to get a functional basic example of zig http support's.

**Review by kassane — COMMENTED** [review]
Need reviewer! (except author)

**Comment by truemedian** [comment]
std.http changed again in <https://github.com/ziglang/zig/pull/15299> (at least this time I have some more complete example code). So this code is already a bit wrong.

I'd like std.http.Server to be considered barely-functional at the moment. It currently leaks memory with every connection and probably has a host of other issues that will be caught when std.http.Client and std.http.Server are tested against eachother, until then I can't advertise much about it's functionality.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -18,8 +18,17 @@ jobs:
...
           version: master
 
-      - name: testing
+      - name: testing (Linux)
+        if: startsWith(matrix.runs-on, 'ubuntu')
         run: sh runall.sh
+      
+      - name: testing (MacOS)
+        if: startsWith(matrix.runs-on, 'macos')
+        run: sh runall.sh
```
Comment (Panquesito7) [inline .github/workflows/CI.yml]: We can shorten the code a little bit here (untested).
```suggestion
      - name: testing (Linux)
        if: (startsWith(matrix.runs-on, 'ubuntu')) || (startsWith(matrix.runs-on, 'macos'))
        run: sh runall.sh
```

**Comment by kassane** [comment]
> std.http changed again in [ziglang/zig#15299](https://github.com/ziglang/zig/pull/15299) (at least this time I have some more complete example code). So this code is already a bit wrong.

New reference update: https://github.com/ziglang/zig/pull/15372

**Comment by nwtgck** [comment]
Hi,

I'm using Zig HTTP server and it works well in 0.11.0-dev.2661+d2a799c65!

* https://github.com/nwtgck/piping-server-zig/blob/3c716bc6d5c19b9abc3b4badf9ad02c093365131/PipingServer.zig
* https://github.com/nwtgck/piping-server-zig/blob/3c716bc6d5c19b9abc3b4badf9ad02c093365131/.github/workflows/ci.yml#L23-L35

I love its API since we can handle in low-level.

**Comment by kassane** [comment]
Finally works!! thx @truemedian @nwtgck

**Review by Panquesito7 — APPROVED** [review]
Great! Thanks. 🚀

File: web/http/client.zig [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+    var h = std.http.Headers{ .allocator = std.testing.allocator };
+    defer h.deinit();
 
-    var req = try client.request(uri, .{}, .{});
+    var req = try client.request(.GET, uri, h, .{});
     defer req.deinit();
-
-    try std.testing.expect(req.response.headers.status == .ok);
+    try req.start(); // start request
+    try req.finish(); // end request
```
Comment (truemedian) [inline web/http/client.zig:13]: This is a noop, there's no payload to finish.

File: web/http/server.zig [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            const server_body: []const u8 = "message from server!\n";
+            res.transfer_encoding = .{ .content_length = server_body.len };
+            try res.headers.append("content-type", "text/plain");
+            try res.headers.append("connection", "close");
+            try res.do();
+
+            var buf: [128]u8 = undefined;
+            const n = try res.readAll(&buf);
+            try expect(std.mem.eql(u8, buf[0..n], "Hello, World!\n"));
+            _ = try res.writer().writeAll(server_body);
```
Comment (truemedian) [inline web/http/server.zig:48]: writeAll returns void, you're discarding nothing here.

File: web/tls/X25519+Kyber768Draft00.zig [hunk 4]
```diff
@@ -2,23 +2,27 @@
...
 
-    var req = try client.request(uri, hdrs, opts);
-    try req.finish();
+    var req = try client.request(.POST, uri, h, .{
+        .header_strategy = .{ .static = &header_buf },
+    });
     defer req.deinit();
-    const read = try req.readAll(&buf);
+    try req.start();
+    try req.finish();
```
Comment (truemedian) [inline web/tls/X25519+Kyber768Draft00.zig:20]: This is a noop, there's no payload to finish.

File: web/http/client.zig (hunk 2 above)
Comment (kassane) [reply web/http/client.zig:13]: Nice. Thanks for the suggestions, I'll make the planned fixes tomorrow.
However, if you want to send a PR, you are welcome.

**Pull request merged**